### PR TITLE
os/filestore: clean up _do_copy_range

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/os/filestore/FileStore.cc [hunk 1]
```diff
@@ -3689,42 +3689,24 @@ int FileStore::_do_copy_range(int from, int to, uint64_t srcoff, uint64_t len, u
...
-	  r = r2;
-	  derr << "FileStore::_do_copy_range: write error at " << pos << "~"
-	    << r-op << ", " << cpp_strerror(r) << dendl;
-
-	  break;
-	}
-	op += (r-op);
-      }
-      if (r < 0)
+      int r = safe_write(to, buf, l);
```
Comment (yuyuyu101) [inline src/os/filestore/FileStore.cc]: shadow variable

**Comment by dachary** [comment]
jenkins test this please (no error logs)

**Comment by dachary** [comment]
jenkins test this please (asok now fixed in master)

**Comment by dachary** [comment]
jenkins test this please (eio now ignored in master)

**Comment by dachary** [comment]
jenkins test this please (eio now ignored in master)

**Comment by dachary** [comment]
jenkins test this please (issue with jenkins agent)

**Comment by tchaikov** [comment]
retest this please.

File: src/os/filestore/FileStore.cc (hunk 1 above)
Comment (tchaikov) [inline src/os/filestore/FileStore.cc]: yeah, i think we can reuse the `r` here without shadowing the `r` in the outer lexical scope. and more importantly, we should note down the `r` and return it if `safe_write()` returns an error. in this change, the inner `r` is logged in the log, and then thrown away `if (r < 0)`.

**Comment by tchaikov** [comment]
retest this please.

**Pull request closed without merging**